### PR TITLE
chore(vector source): Rate limit TCP connection errors, downgrade their severity

### PR DESCRIPTION
**Pull request by Hoverbear**
Fixes https://github.com/timberio/vector/issues/2664 by:

* Rate limiting `TCP Connection Error` logs
* Reduces their severity to `INFO` from `WARN` (Issue suggested `DEBUG`, looking for consensus here)

### CODE REVIEW

**Comment by binarylogic** [comment]
Why info?

**Comment by Hoverbear** [comment]
I don't really have a strong opinion on that one. I'm worried debug won't be noticed by most people, and warn will be too alert-y for others. Do you have dogmatic guidance? :)

**Comment by binarylogic** [comment]
I think about it like:

* `warn` - anything exceptional that does not halt the processing of data.
* `error` - anything that halts the processing of data
* `fatal` - (if we even have this) a Vector exception. Something that crashes Vector.

This should be on the `warn`, IMO.

File: src/internal_events/tcp.rs [hunk 1]
```diff
@@ -80,7 +80,7 @@ pub struct TcpConnectionError {
 
 impl InternalEvent for TcpConnectionError {
     fn emit_logs(&self) {
-        warn!(message = "connection error.", error = %self.error);
+        info!(message = "connection error.", error = %self.error, rate_limit_secs = 10);
```
Comment (bruceg) [inline src/internal_events/tcp.rs:83]: I'm :+1: on the rate limit, but I don't see why it should be downgraded to info.

Comment (Hoverbear) [reply src/internal_events/tcp.rs:83]: Ok, agree!

**Pull request merged**